Unity-C#/README.md

```diff
@@ -0,0 +1,252 @@
+
+The nominal case should be put in the "if" part and the exception in the "else" part of an "if" statement.
+
+Executable statements in conditionals MUST be avoided.
+
+### Miscellaneous
+The use of magic numbers in the code should be avoided; they should be declared using named **constants** instead.
+
+
+## Object and Array creation
+```csharp
+
+```
+
+
+## Conditions
+Any non-trivial conditions should be assigned to a descriptive variable
+```csharp
+bool isAuthorized = (user.IsAdmin() || user.IsModerator());
+if (isAuthorized)
+{
+	Console.Log('winning');
+}
+```
+
+## Return statement
+To avoid deep nesting of if-statements, always return a function's value as early as possible.
+
+```csharp
+public bool IsPercentage(int percent) 
+{
+	if (percent < 0)
+	{
+		return false;
+	}
+
+	if (percent > 100)
+	{
+		return false;
+	}
+
+	return true;
+}
+
+```
+
+
+## Comments
+
+Comments are the documentation of your code; therefore **tricky code should not be commented but rewritten**. 
+
+All comments should be written in English and be politically correct, even those that are supposed to be temporary.
+
+Comments should be indented relative to their position in the code, preceding the code in question.
+
+Even if everybody likes commented code, do not comment every single line of code; you will kill the readability of it.
+
+Comments are supposed to explain an algorithm, not repeat it in a different language. This is wrong:
+```csharp
+// increment i by 1
+i += 1;
+```
+
+Try to put comment at the top of a block of code to explain the point of it..
+
+
+### Comment Syntax
+The Visual Studio syntax is based on XML . Many tools can extract this XML and create useful help documents for you. These comments must be well-formed.
+
+It is important to note that projects in Unity may not be using Visual Studio, in this case one style should be picked, and be accepted project wide.
+```csharp
+/// <summary>
+/// You can use the summary tags above any piece of code so tools can extract this
+/// </summary>
+
+// If you do not want the comment to appear you can use the normal comment syntax,
+// the use of /* comment */ is discouraged.
+
+// Single line version is preferred
+```
+
+### Class Comments
+Classes must be documented with a description summary
+```csharp
+/// <summary>
+/// A simple user class
+/// </summary>
+public class User
+{
+	/// <summary>
+	/// User constructor initialises user name
+	/// </summary>
+	/// <param name="name">Persons name</param>
+	public User(string name) 
+	{
+		// ...
+	}
+}
+
+// A simple user class
+public class User
+{
+	// User constructor initialises user name
+	// name - Persons name
+	public User(string name) 
+	{
+		// ...
+	}
+}
+```
+
+### Method Comments
+Method descriptions should start with a sentence written in the third person declarative voice.
+```csharp
+/// <summary>
+/// Renames the user
+/// </summary>
+/// <param name="name">Persons name</param>
```

Comment:
The old name or the new name? Also "Person's" or "The person's". You also mix "user" and "person".

Suggestion for this param description: "The new name to assign to the user"